### PR TITLE
Feature/alert for checker

### PR DESCRIPTION
**Pull request by duyanguang**
add alert function, add caller file and method to message

### CODE REVIEW

File: lyrebird/event.py [hunk 1]
```diff
@@ -121,5 +124,14 @@ def unregister(self, event_bus):
...
\ No newline at end of file
+        application.server['event'].publish(channel, message, *args, **kwargs)
+
+    def alert(self, channel, message, *args, **kwargs):
+        stack = inspect.stack()
+        caller_file = stack[1].filename
+        self.report_file = caller_file[caller_file.rfind('/') + 1:]
+        self.report_method = stack[1].function
+        if isinstance(message, dict):
+            message['report_file'] = self.report_file
```
Comment (zhaoye) [inline lyrebird/event.py]: Could we call it  "script_name"

File: lyrebird/event.py [hunk 2]
```diff
@@ -121,5 +124,14 @@ def unregister(self, event_bus):
...
+            message['report_method'] = self.report_method
```
Comment (zhaoye) [inline lyrebird/event.py]: And could we name it as "function_name" ?

File: lyrebird/event.py [hunk 3]
```diff
@@ -105,6 +106,8 @@ def on_message(data):
     """
     def __init__(self):
         self.listeners = []
+        self.report_file = ''
```
Comment (zhaoye) [inline lyrebird/event.py]: Remove unused var plz

File: lyrebird/event.py [hunk 4]
```diff
@@ -105,6 +106,8 @@ def on_message(data):
...
+        self.report_method = ''
```
Comment (zhaoye) [inline lyrebird/event.py]: Remove unused var plz

File: tests/test_event.py [hunk 5]
```diff
@@ -49,10 +51,32 @@ def test_state(event_server):
...
     test_state = lyrebird.state.get('Test')
     assert test_state == None
     lyrebird.publish('Test', 'TestMessage', state=True)
     test_state = lyrebird.state.get('Test')
     assert test_state == 'TestMessage'
+
+
+def test_customer_event_alert(event_server):
+    custom_event = CustomEventReceiver()
+    lyrebird.application.server['event'] = event_server
```
Comment (zhaoye) [inline tests/test_event.py]: This event server already add into "application.server"
It not necessary to add it again here.

File: lyrebird/event.py (hunk 2 above)
Comment (zhaoye) [inline lyrebird/event.py]: And could we name it as "function_name" ?

File: tests/test_event.py [hunk 6]
```diff
@@ -49,10 +51,32 @@ def test_state(event_server):
...
+                    'plugin': 'perf.cpu'
+                })
+
+    assert custom_event.report_method == 'test_customer_event_alert'
+    assert custom_event.report_file[custom_event.report_file.rfind('/') + 1:] == 'test_event.py'
+
+
+def test_customer_event_alert_not_dict(event_server):
+    custom_event = CustomEventReceiver()
+    lyrebird.application.server['event'] = event_server
```
Comment (zhaoye) [inline tests/test_event.py]: Remove this line, plz

File: lyrebird/event.py (hunk 2 above)
Comment (duyanguang) [reply lyrebird/event.py]: Good idea, will accept it and change the name

**Pull request merged**